password.go

```diff
@@ -17,9 +20,49 @@ var PasswordQuestionTemplate = `
 {{- color "green+hb"}}? {{color "reset"}}
 {{- color "default+hb"}}{{ .Message }} {{color "reset"}}`
 
+type filteredReadlinePasswordOutput struct {
+	len int
+}
+
+// Write will be called by readline to echo the masked characters to the console.  It assumes
+// full control of the output with prompting, but since survey is printing the prompt readline
+// does not play nice here.  Write is called with clear-line statements which need to be
+// ignored then it is called with the full buffered output. So if you type "hi" this is the
+// sequence that will be sent to Write
+// 0: "\033[J\033[2K\r"
+// 1: " \b"
+// 2: "\033[J\033[2K\r"
+// 3: "*" (this is the masked "h" from "hi")
+// 4: "\033[J\033[2K\r"
+// 5: "**" (this is the masked "hi" from "hi")
+func (o *filteredReadlinePasswordOutput) Write(p []byte) (n int, err error) {
```

Comment:
Hmm so I still have to go through this a bit more but I wonder if it's better to just leave password as sort of an odd man and remove the `renderer` from its implementation. There is a massive win in unifying all of the re-rendering logic but I fear this might cause more bugs than the pain in having to potentially change two different places for rendering logic.